packages/infra/lib/api-stack/api-service.ts

```diff
@@ -133,15 +135,18 @@ export function createAPIService({
   const listenerPort = 80;
   const containerPort = 8080;
   const logGroup = LogGroup.fromLogGroupArn(stack, "ApiLogGroup", props.config.logArn);
+  const desiredTaskCount = isProd(props.config) ? 6 : isSandbox(props.config) ? 2 : 1;
+  const maxTaskCount = isProd(props.config) ? 40 : isSandbox(props.config) ? 10 : 5;
   const fargateService = new ecs_patterns.ApplicationLoadBalancedFargateService(
     stack,
     "APIFargateServiceAlb",
     {
       cluster: cluster,
-      // Watch out for the combination of vCPUs and memory, more vCPU requires more memory
+      // Watch out for the combination of vCPUs and memory.
       // https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task_definition_parameters.html#task_size
-      cpu: isProd(props.config) ? 2048 : 1024,
-      desiredCount: isProd(props.config) ? 2 : 1,
+      cpu: 1024, // Keep to 1 vCPU because NodeJS is single-threaded
+      memoryLimitMiB: isProd(props.config) ? 2048 : 1024,
```

Comment:
- prod: we're currently running under 30% CPU for 3 weeks ([source](https://us-west-1.console.aws.amazon.com/cloudwatch/home?region=us-west-1#metricsV2?graph=~(metrics~(~(~'AWS*2fECS~'MemoryUtilization~'ServiceName~'APIInfrastructureStack-APIFargateServiceAlbService1982B8ED-lHbszm7GZIg5~'ClusterName~'APIInfrastructureStack-APICluster381F6E68-v0y03IsPXhdE~(stat~'Minimum~region~'us-west-1~visible~false))~(~'...~(stat~'Maximum~region~'us-west-1))~(~'...~(stat~'Average~region~'us-west-1~visible~false)))~period~3600~region~'us-west-1~stacked~false~title~'Memory*20utilization~view~'timeSeries~start~'-PT672H~end~'P0D))), so it should be safe to halve it
- sandbox: we're under 15% CPU for the same period ([source](https://us-west-2.console.aws.amazon.com/cloudwatch/home?region=us-west-2#metricsV2?graph=~(metrics~(~(~'AWS*2fECS~'MemoryUtilization~'ServiceName~'SandboxAPIInfrastructureStack-APIFargateServiceAlbService1982B8ED-fjrZbgBHqkou~'ClusterName~'SandboxAPIInfrastructureStack-APICluster381F6E68-CIoumzbxHLSJ~(stat~'Minimum~region~'us-west-2~visible~false))~(~'...~(stat~'Maximum~region~'us-west-2))~(~'...~(stat~'Average~region~'us-west-2~visible~false)))~period~3600~region~'us-west-2~stacked~false~title~'Memory*20utilization~view~'timeSeries~start~'-PT672H~end~'P0D)))

Reply:
Let's leave it as it for first iteration but make a google calendar event to look at the system in a week to adjust lower (or if there's a master issue, put it there).

Reply:
yeah why not leave memory the same? it can get spiky with larger consolidated payloads 


are we limited by the cpu?

Reply:
Yeah, we decided to leave memory the same - [context](https://metriport.slack.com/archives/C04DMKE9DME/p1720827000931059?thread_ts=1720657415.437769&cid=C04DMKE9DME).

Reply:
updated